src/httpClient/deduplicateRequestAdapter.ts

```diff
@@ -0,0 +1,24 @@
+import { AxiosAdapter, AxiosPromise } from 'axios';
+
+// adopted from https://github.com/RasCarlito/axios-cache-adapter/issues/231#issuecomment-880288436
```

Comment:
I like how someone there called this "debouncing". I think it could be a more descriptive name than deduplication?

Reply:
Agreed, it is more precise. I have renamed all occurrences.